cleo_sdk/CLEO_Utils.h

```diff
@@ -214,6 +214,22 @@ namespace CLEO
         return _paramsArray[0];
     }
 
+    static SCRIPT_VAR& _readParamFloat(CRunningScript* thread)
+    {
+        auto var = _readParam(thread);
+
+        // people tend to use '0' instead '0.0' when providing literal float params in scripts
+        // binary these are equal, so can be allowed
+        if (var.dwParam == 0)
```

Comment:
This is already existing problem as valid until today scripts throws critical errors because someone did
`setPos 0 0 0` instead of `setPos 0.0 0.0 0.0`. Sanny did not carred and compiled it as BYTE instead of FLOAT.

I have not yet met case where somebody did `setPos 0x3f800000 0x3f800000 0x3f800000 `

Reply:
my suggestion would be to allow any int values in place of floats. values like `0` or `0x3f800000` can work out perfectly fine.

the opposite should be disallowed. using floats in place of integers is almost always a mistake (maybe except 0.0, however `is_player_playing 0.0` makes little sense)

Reply:
1 is not binary equal to the 1.0

Reply:
I dunno, I don't see a point in made-up limitations. If someone wants to write `setPos 0x3f800000 0x3f800000 0x3f800000` what's bad about it? It is valid and working code, why should we limit it?

Reply:
Nobody writes that. But usually people write `setPos 1.0 1.0 1` which results in float like 1.4E-50

Reply:
We can't use "nobody writes that" argument. there could be an infinite number of scripts and some of them may use such code for some reason.

I agree that `1` is a likely an overlook the user should be warned about. However, we also reject whole class of valid values which work with original values, but not with CLEO ones.

at some point there should be a switch to disable some or all these restrictions. Possibly something like debug/release builds.